specification/_types/Geo.ts

```diff
@@ -56,6 +56,14 @@ export enum GeoDistanceType {
 /** A GeoJson shape, that can also use Elasticsearch's `envelope` extension. */
 export type GeoShape = UserDefinedValue
 
+/** A GeoJson GeoLine. */
+export class GeoLine {
+  /** Always `"LineString"` */
+  type: string
```

Comment:
If it's always `LineString`, why not define it as is? Or use a single element enum?

Reply:
In a previous iteration I defined it as a literal `type: 'LineString'`, but this has some implications on code generators that would now have to implement code generation for arbitrary constant values.

Our other uses of literal values is for union tags (`@variants internal tag=xxx`), which are handled in a very specific way.

Since this type is used as part of a response in the `geo_line` aggregation, defining it as `string` should be fine since users normally never have to write it in their applications.